tests/test_regression.py

```diff
@@ -0,0 +1,124 @@
+"""
+    Suite of tests that go through the entire process of creating a new fktable
+    from a empty folder.
+
+    The target theory is 400 and the relevant `.toml`, theory runcard and eko template
+    are downloaded from https://github.com/NNPDF/theories during this test so this tests
+    has the double effect of ensuring compatibility between both repositories.
+"""
+from pathlib import Path
+from subprocess import run
+from urllib.request import urlretrieve
+
+import numpy as np
+import pytest
+from pineappl.fk_table import FkTable
+from yaml import dump, safe_load
+
+THEORIES_REPO = "https://raw.githubusercontent.com/NNPDF/theories/main"
+THEORYID = 400
+
+
+def _download_resources(filename, tmp_path):
+    """Download resources (filename) from the theories repo and put it in the same path
+    relative to THEORIES_REPO
+    """
+    output_file = tmp_path / filename
+    output_file.parent.mkdir(exist_ok=True)
+    urlretrieve(f"{THEORIES_REPO}/{filename}", output_file)
```

Comment:
In principle, URLs are like paths, at least the way you're using they are actually `PurePosixPath`s, since they are locations inside `THEORIES_REPO` (they are pure because you can not operate on them, e.g. you can not ask if a file exists, or create a directory).

If you wish, you could use that, as a better abstraction over strings.